workspace/notebook/py_sb_horizon_harmonised_appeal_has.json

```diff
@@ -0,0 +1,857 @@
+					"from pyspark.sql import functions as F, Window as W\r\n",
+					"from pyspark.sql import Row\r\n",
+					"from pyspark.sql.functions import *\r\n",
+					"from datetime import date,datetime\r\n",
+					"import builtins"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"##### Initialise Logging decorator"
+				]
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"%run utils/py_logging_decorator"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"##### Initialise Application Insight Logging functions"
+				]
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"%run utils/py_utils_common_logging_output"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "code",
+				"source": [
+					"%run utils/py_applicationinsights"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"#### Config"
+				]
+			},
+			{
+				"cell_type": "code",
+				"metadata": {
+					"jupyter": {
+						"source_hidden": false,
+						"outputs_hidden": false
+					},
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"# Tables\r\n",
+					"service_bus_table = \"odw_harmonised_db.sb_appeal_has\"\r\n",
+					"horizon_table     = \"odw_standardised_db.horizon_appeal_has\"\r\n",
+					"spark_table_final = \"odw_harmonised_db.appeal_has\"\r\n",
+					"\r\n",
+					"# Keys\r\n",
+					"primary_key = 'caseReference'\r\n",
+					"incremental_key = 'AppealsHasID'\r\n",
+					"\r\n",
+					"# Run context\r\n",
+					"start_exec_time = datetime.now()\r\n",
+					"end_exec_time = start_exec_time\r\n",
+					"insert_count = 0\r\n",
+					"update_count = 0\r\n",
+					"delete_count = 0\r\n",
+					"error_message=''"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "markdown",
+				"metadata": {
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"##### Read Service Bus (already in schema)"
+				]
+			},
+			{
+				"cell_type": "code",
+				"metadata": {
+					"jupyter": {
+						"source_hidden": false,
+						"outputs_hidden": false
+					},
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"if not error_message:\r\n",
+					"    try:\r\n",
+					"        # Get data out of the service bus with additional fields needed for Horizon data\r\n",
+					"        service_bus_data = spark.sql(f\"\"\"\r\n",
```

Comment:
I think this might be neater if written entirely using pyspark rather than sql, but not a big issue

Reply:
Yes it could - we just followed the same pattern as it was in s78 before I updated it